qt/scientific_interfaces/ISISReflectometry/test/Batch/RowProcessingAlgorithmTest.h

```diff
@@ -49,6 +49,14 @@ class RowProcessingAlgorithmTest : public CxxTest::TestSuite {
     checkExperimentSettings(*result);
   }
 
+  void testExperimentSettingsReductionTypeSetToNormalForSumInLambda() {
+    auto model = Batch(makeExperimentWithReductionTypeSetForSumInLambda(), m_instrument, m_runsTable, m_slicing);
```

Comment:
The test failures are caused by passing a temporary variable in the constructor of the `Batch`. This would be the right thing to do if passing by value, but the Batch constructor takes a reference, so when the temporary gets deleted, the Batch ends up holding a reference to something that doesn't exist any more. The error message about an incorrect analysis mode is just because that's the first part of the `Experiment` it attempts to use and it contains junk data.

This is easily fixed by creating a local variable the Batch can take a reference to instead for the duration of the test:

```suggestion
    auto experiment = makeExperimentWithReductionTypeSetForSumInLambda();
    auto model = Batch(experiment, m_instrument, m_runsTable, m_slicing);
```